builtin/game/misc.lua

```diff
@@ -168,6 +168,18 @@ function core.setting_get_pos(name)
 	return core.string_to_pos(value)
 end
 
+function core.get_world_limits()
+	local absolute_limit = 31000
+	local value = core.setting_get("map_generation_limit")
+	local v
+	if not value or value > absolute_limit then
+		v = vector.new(absolute_limit, absolute_limit, absolute_limit)
+	else
+		v = vector.new(value, value, value)
+	end
+	return vector.subtract(0, v), v
```

Comment:
No. Mods can create structures that extend beyond 31000, the absolute limit is the limits of s16 co-ordinates.
Also, 'mapgen limit' can be, and is sometimes, reduced, but leaves world present outside the new limit. Mods can also create structures that extend beyond the 'mapgen limit' setting.